### PR TITLE
Add some framework stuff to the project

### PR DESCRIPTION
**Pull request by hawkowl**
cc @glyph

### CODE REVIEW

**Comment by glyph** [comment]
ron-paul

File: src/warpedalloy/__init__.py [hunk 1]
```diff
@@ -35,12 +34,12 @@
 @attr.s
 class SendToSubprocess(Protocol, object):
     """
-    
+
```
Comment (glyph) [inline src/warpedalloy/__init__.py]: It would probably be better to just delete the empty docstrings entirely...

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+*.py[co]
+*.so
+*.DS_Store
+_trial_temp*/
+build/
+dropin.cache
+doc/
+docs/_build/
+dist/
+venv/
```
Comment (glyph) [inline .gitignore]: What creates ./venv?

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,19 @@
+*.egg-info/
+.eggs/
+*.o
+*.py[co]
+*.so
+*.DS_Store
+_trial_temp*/
+build/
+dropin.cache
+doc/
```
Comment (glyph) [inline .gitignore]: I'm pretty sure `doc/` shouldn't be ignored?

Comment (hawkowl) [reply .gitignore]: `docs/` shouldn't, but `doc/` is where Lore used to stuff docs. This is copied from Twisted :)

File: .gitignore (hunk 2 above)
Comment (hawkowl) [reply .gitignore]: me

Comment (glyph) [reply .gitignore]: If you make a dir called `venv` personally, then you should probably put something in your git global-ignore; we should only put stuff in here that the project itself, or specific recommended tooling (tox, trial, etc) creates.

File: .gitignore (hunk 3 above)
Comment (glyph) [reply .gitignore]: We should probably get rid of it there too, then :)

Comment (hawkowl) [reply .gitignore]: You mean I shouldn't just cargo cult this append only `.gitignore` into various projects? ;)

File: src/warpedalloy/__init__.py (hunk 1 above)
Comment (hawkowl) [reply src/warpedalloy/__init__.py]: Yeah, that's a good idea

File: .gitignore (hunk 3 above)
Comment (hawkowl) [reply .gitignore]: Done

File: src/warpedalloy/__init__.py [hunk 4]
```diff
@@ -219,13 +218,13 @@ def sendOutFileDescriptor(self, fileDescriptor):
...
         """
-        
+
         """
         here, there = socketpair(AF_UNIX, SOCK_STREAM)
         owp = OneWorkerProtocol()
         serverTransport = UNIXServer(here, owp, None, None, 4321, self.reactor)
         owp.makeConnection(serverTransport)
-        argv = [sys.executable, __main__.__file__, b'w']
+        argv = [sys.executable, __file__, b'w']
```
Comment (glyph) [inline src/warpedalloy/__init__.py]: Why'd you change this from `__main__.__file__`, which re-runs the script that was run, to just `__file__`, which changes it to this file?

File: .gitignore (hunk 2 above)
Comment (hawkowl) [reply .gitignore]: Done

File: src/warpedalloy/__init__.py (hunk 1 above)
Comment (hawkowl) [reply src/warpedalloy/__init__.py]: Done

File: src/warpedalloy/__init__.py (hunk 4 above)
Comment (hawkowl) [reply src/warpedalloy/__init__.py]: oh _that's_ what this code does

Comment (hawkowl) [reply src/warpedalloy/__init__.py]: Done :D

**Comment by glyph** [comment]
yolo

**Pull request merged**